source/uwp/Renderer/lib/MediaHelpers.cpp

```diff
@@ -179,24 +179,97 @@ namespace AdaptiveCards::Rendering::Uwp::MediaHelpers
         return {mediaSourceUrl, mimeType};
     }
 
+    void SetMediaSourceHelper(winrt::MediaPlayerElement const& mediaElement,
+                              winrt::AdaptiveMedia const& adaptiveMedia,
+                              winrt::AdaptiveRenderContext const& renderContext,
+                              winrt::MediaSource const& mediaSrc)
+    {
+        if (adaptiveMedia.CaptionSources().Size() > 0)
+        {
+            for (const auto captionSource : adaptiveMedia.CaptionSources())
+            {
+                const auto currentCaptionType = captionSource.MimeType();
+                for (auto&& supportedCaptionType : supportedCaptionTypes)
+                {
+                    if (currentCaptionType == supportedCaptionType)
+                    {
+                        const auto timedTextURL = GetUrlFromString(renderContext.HostConfig(), captionSource.Url());
+
+                        winrt::IAdaptiveCardResourceResolver resourceResolver{nullptr};
+                        if (const auto resourceResolvers = renderContext.ResourceResolvers())
+                        {
+                            resourceResolver = resourceResolvers.Get(timedTextURL.SchemeName());
+                        }
+
+                        const auto timedTextSrcResolvedHelper = [captionSource](winrt::TimedTextSource const& /*sender*/,
+                                                                                winrt::TimedTextSourceResolveResultEventArgs const& args)
+                                {
+                                    if (!args.Error())
+                                    {
+                                        args.Tracks().GetAt(0).Label(captionSource.Label());
+                                    }
+                                };
+                        if (resourceResolver == nullptr)
+                        {
+                            const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
+                            timedTextSrc.Resolved(timedTextSrcResolvedHelper);
+                            mediaSrc.ExternalTimedTextSources().Append(timedTextSrc);
+                        }
+                        else
+                        {
+                            auto args = winrt::make<winrt::implementation::AdaptiveCardGetResourceStreamArgs>(timedTextURL);
+                            auto getResourceStreamOperation = resourceResolver.GetResourceStreamAsync(args);
+                            getResourceStreamOperation.Completed(
+                                [mediaSrc, captionSource, timedTextSrcResolvedHelper](winrt::IAsyncOperation<winrt::IRandomAccessStream> operation,
```

Comment:
probably no need to capture `captionSource` anymore :)

Reply:
Yep, thanks for catching that